builder/dockerignore/dockerignore.go

```diff
@@ -33,3 +36,75 @@ func ReadAll(reader io.ReadCloser) ([]string, error) {
 	}
 	return excludes, nil
 }
+
+// ReadAllRecursive takes a base directory and a starting directory,
+// reads all .dockerignore files in subdirectories and returns the list of file
+// patterns to ignore. Note this will trim whitespace from each line as well
+// as use GO's "clean" func to get the shortest/cleanest path for each.
+func ReadAllRecursive(baseDir string, directory string) ([]string, error) {
```

Comment:
Without looking at the code (I just looked at the description), will this always do a deep scan for `.dockerignore` files? Thinking of this situation, where a deep scan is not needed;

```
cat ./.dockerignore
*
```

In this case, the `.dockerignore` at the root of the build-context excludes all files, so no scanning for other `.dockerignore` files should be needed.